Library/Homebrew/extend/pathname.rb

```diff
@@ -436,11 +436,12 @@ def env_script_all_files(dst, env)
   end
 
   # Writes an exec script that invokes a java jar
-  def write_jar_script(target_jar, script_name, java_opts = "")
+  def write_jar_script(target_jar, script_name, java_opts = "", java_version = "")
     mkpath
+    launcher = java_version.empty? ? "java" : "/usr/libexec/java_home -v #{java_version} --exec java"
```

Comment:
After speaking to @ilovezfs instead let's do something like:

```ruby
java_home = if java_version
  <<~EOS
    export JAVA_HOME="$(#{Language::Java.java_home_cmd(java_version)})"

  EOS
end
```

and below:

```bash
       #!/bin/bash
       #{java_home}exec java #{java_opts} -jar #{target_jar} "$@"
```

This will be consistent with what `bin.env_script_all_files(libexec/"bin", Language::Java.java_home_env("1.8"))` ends up doing.

Reply:
If my understanding is correct, this only specifies `JAVA_HOME`, hence still relies on an unpolluted `$PATH`, if `$PATH` provides another `/usr/local/java-1.7/bin/java`, setting `JAVA_HOME` only no longer has the magic, and will launch java 1.7 instead, as I presented in my comment above: https://github.com/Homebrew/brew/pull/3656#issuecomment-359259313

Reply:
We'd rather be consistent between approaches. Let's go for this one in both places for now and then if desired you can submit an update for both later.